### PR TITLE
feat: Component protocol default protocol yml support i18n

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Component protocol default protocol yml support i18n

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reivewers:
/assign @sfwn

### CODE REVIEW

File: providers/component-protocol/protocol.service.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (s *protocolService) Render(ctx context.Context, req *pb.RenderRequest) (*p
 	sdk.Ctx = ctx
 
 	// render concrete scenario
-	if err := protocol.RunScenarioRender(ctx, renderReq); err != nil {
+	if err := protocol.RunScenarioRender(ctx, renderReq, sdk.Tran); err != nil {
```
Comment (sfwn) [inline providers/component-protocol/protocol.service.go:61]: get tran from ctx.sdk

File: providers/component-protocol/protocol/protocol.go [hunk 2]
```diff
@@ -85,12 +90,33 @@ func RegisterDefaultProtocolsFromBasePath(basePath string) {
...
 
 // getDefaultProtocol get default protocol by scenario.
-func getDefaultProtocol(ctx context.Context, scenario string) (cptype.ComponentProtocol, error) {
-	s, ok := defaultProtocols[scenario]
+func getDefaultProtocol(ctx context.Context, scenario string, tran pi18n.Translator) (cptype.ComponentProtocol, error) {
+	lang := cputil.Language(ctx)
+	if lang.Len() == 0 {
+		c, ok := defaultProtocols[scenario]
+		if !ok {
+			return cptype.ComponentProtocol{}, fmt.Errorf(i18n(ctx, "${default.protocol.not.exist}, ${scenario}: %s", scenario))
```
Comment (sfwn) [inline providers/component-protocol/protocol/protocol.go:98]: dup with L104

File: providers/component-protocol/protocol/protocol.go [hunk 3]
```diff
@@ -85,12 +90,33 @@ func RegisterDefaultProtocolsFromBasePath(basePath string) {
...
+		}
+		return c, nil
+	}
+	s, ok := defaultProtocolsRaw[scenario]
 	if !ok {
 		return cptype.ComponentProtocol{}, fmt.Errorf(i18n(ctx, "${default.protocol.not.exist}, ${scenario}: %s", scenario))
 	}
-	return s, nil
+	for _, v := range CpRe.FindAllStringSubmatch(s, -1) {
+		if len(v) == 2 && strings.HasPrefix(v[1], I18n+".") {
```
Comment (sfwn) [inline providers/component-protocol/protocol/protocol.go:107]: use strutil.ReplaceAllStringSubmatchFunc()

**Review by sfwn — CHANGES_REQUESTED** [review]

File: providers/component-protocol/protocol/protocol.go [hunk 4]
```diff
@@ -41,6 +45,7 @@ func RegisterDefaultProtocols(protocolYAMLs ...[]byte) {
 			continue
 		}
 		defaultProtocols[p.Scenario] = p
+		defaultProtocolsRaw[p.Scenario] = string(protocolYAML)
```
Comment (sfwn) [inline providers/component-protocol/protocol/protocol.go:48]: add only when yaml match cp placeholder regex for performance consider.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: providers/component-protocol/protocol/protocol.go [hunk 5]
```diff
@@ -86,11 +93,31 @@ func RegisterDefaultProtocolsFromBasePath(basePath string) {
...
 // getDefaultProtocol get default protocol by scenario.
 func getDefaultProtocol(ctx context.Context, scenario string) (cptype.ComponentProtocol, error) {
-	s, ok := defaultProtocols[scenario]
+	lang := cputil.Language(ctx)
+	tran := ctx.Value(cptype.GlobalInnerKeyCtxSDK).(*cptype.SDK).Tran
+	s, ok := defaultProtocolsRaw[scenario]
 	if !ok {
-		return cptype.ComponentProtocol{}, fmt.Errorf(i18n(ctx, "${default.protocol.not.exist}, ${scenario}: %s", scenario))
+		// protocol not have cp placeholder
+		s, ok := defaultProtocols[scenario]
```
Comment (sfwn) [inline providers/component-protocol/protocol/protocol.go:101]: use `p` uniformly as before

**Review by sfwn — CHANGES_REQUESTED** [review]

File: providers/component-protocol/protocol/protocol.go (hunk 5 above)
Comment (shuofan) [reply providers/component-protocol/protocol/protocol.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**